sm/download/src/lib.rs

```diff
@@ -0,0 +1,3 @@
+pub mod c8y;
```

Comment:
One should find a way to move all the code specific to Cumulocity elsewhere in some `common/cumulocity` crate.

* That c8y crate would provide all the methods to get an external id for a device id, a JWT token from the MQTT endpoint, and to build any c8y specific urls.
* The download crate will assume that all the urls and credentials are given ready to be used.
* The glue is to be made by the c8y-sm-mapper. The latter is c8y aware, can build any specific urls and can retrieve the JWT token. These urls+credentials are then forwarded to the agent that can operate the download.

The point is to have:
* The agent cloud agnostic (hence unaware of how the urls and credentials can be specific to a cloud endpoint). 
* The mappers state less (hence not responsible of the downloads).